lnrpc/routerrpc/router_backend.go

```diff
@@ -858,6 +858,12 @@ func (r *RouterBackend) extractIntentFromSendRequest(
 	}
 	payIntent.DestCustomRecords = customRecords
 
+	firstHopRecords := record.CustomSet(rpcPayReq.FirstHopCustomRecords)
+	if err := firstHopRecords.Validate(); err != nil {
+		return nil, err
+	}
+	payIntent.FirstHopCustomRecords = firstHopRecords
```

Comment:
Looks like the `FirstHopCustomRecords` field was added in a later commit. I think this one is more temp, but if not, we should move the commit that adds the new field to before this one.